src/session-managers/SecureSessionManager.ts

```diff
@@ -33,13 +34,17 @@ export class SecureSessionManager {
     }
   }
 
-  async logIn(request: FastifyRequest, user: any) {
+  async logIn(request: FastifyRequest, user: any, options?: AuthenticateOptions) {
     const object = await this.serializeUser(user, request)
 
     // Handle @fastify/session to prevent token/CSRF fixation
     if (request.session.regenerate) {
       if (this.clearSessionOnLogin && object) {
-        await request.session.regenerate(this.clearSessionIgnoreFields)
+        const keepSessionInfoKeys: string[] = [...this.clearSessionIgnoreFields]
+        if (options?.keepSessionInfo) {
+          keepSessionInfoKeys.push(...(Object.keys(request.session) || []))
```

Comment:
nit: Object.keys cannot return a falsy value, no?

```suggestion
          keepSessionInfoKeys.push(...(Object.keys(request.session)))
```